### PR TITLE
feat/teaming: Added revamped UI of teaming for litmus 3.0.0

### PR DESCRIPTION
**Pull request by Saranya-jena**
## Proposed changes

image
image
image

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

**Review by SahilKr24 — APPROVED** [review]
Looks good! Great improvement over last time. :rocket:

File: chaoscenter/web/src/api/entities/project.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  invitation: Invitation;
+  joinedAt: string;
+}
+
+export interface Project {
+  projectID: string;
+  name: string;
+  members: Members[];
+  UID: string;
+}
```
Comment (hrishavjha) [inline chaoscenter/web/src/api/entities/project.ts:24]: Are these two interfaced being used anywhere?

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { generateActiveMemberTableContent } from './helper';
+
+export default function ActiveProjectMembersController(): React.ReactElement {
+  const { projectID } = useParams<{ projectID: string }>();
+  const [projectMembers, setProjectMembers] = React.useState<ProjectMember[]>([]);
+  const {
+    data,
+    isLoading,
+    refetch: getMembersRefetch
+  } = useGetProjectMembersQuery({ project_id: projectID, state: 'accepted' });
```
Comment (hrishavjha) [inline chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx:14]: Add these values in enum and consume it here.

File: chaoscenter/web/src/controllers/PendingProjectMemberList/PendingProjectMembers.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { ProjectMember, useGetProjectMembersQuery } from '@api/auth';
+
+export default function PendingProjectMembersController(): React.ReactElement {
+  const { projectID } = useParams<{ projectID: string }>();
+  const [projectMembers, setProjectMembers] = React.useState<ProjectMember[]>([]);
+  const {
+    data,
+    isLoading,
+    refetch: getPendingMembersRefetch
+  } = useGetProjectMembersQuery({ project_id: projectID, state: 'not_accepted' });
```
Comment (hrishavjha) [inline chaoscenter/web/src/controllers/PendingProjectMemberList/PendingProjectMembers.tsx:14]: add to enum.

**Review by hrishavjha — APPROVED** [review]

File: chaoscenter/graphql/server/api/middleware/logger.go [hunk 4]
```diff
@@ -56,9 +56,10 @@ func StructuredLogger(logger *logrus.Logger) gin.HandlerFunc {
 
 		if c.Writer.Status() >= 500 {
 			logrus.WithFields(logFields).Error()
-		} else {
-			logrus.WithFields(logFields).Info()
 		}
+		//} else {
+		//	logrus.WithFields(logFields).Info()
+		//}
```
Comment (arkajyotiMukherjee) [inline chaoscenter/graphql/server/api/middleware/logger.go:62]: commented code

File: chaoscenter/web/src/api/entities/project.ts (hunk 1 above)
Comment (arkajyotiMukherjee) [reply chaoscenter/web/src/api/entities/project.ts:24]: it's always good to type them all out @hrishavjha

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx (hunk 2 above)
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx:14]: Invitation.ACCEPTED has a value of 'Accepted' but this requires 'accepted'
Can we make it the same...follow one convention

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/helper.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    memberData && memberData?.length > 0
+      ? memberData.map(member => {
+          return {
+            userID: member.userID,
+            name: member.name,
+            username: member.username,
+            email: member.email,
+            invitation: member.invitation,
+            joinedAt: member.joinedAt,
+            role: member.role
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/ActiveProjectMemberList/helper.ts:14]:
```suggestion
            ...member
```

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/helper.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import type { ProjectMember } from '@api/auth';
+
+export function generateActiveMemberTableContent(memberData: Array<ProjectMember> | undefined): Array<ProjectMember> {
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/ActiveProjectMemberList/helper.ts:3]: I don't understand the utility of this function...it's not doing any modification on memberData

File: chaoscenter/web/src/controllers/InviteNewMembers/InviteNewMembers.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import InviteNewMembersView from '@views/InviteNewMembers/InviteNewMembers';
+import { generateInviteUsersTableContent } from './helper';
+import type { InviteUserDetails } from './types';
+
+interface InviteUsersControllerProps {
+  handleClose: () => void;
+}
+
+export default function InviteUsersController({ handleClose }: InviteUsersControllerProps): React.ReactElement {
+  const { projectID } = useParams<{ projectID: string }>();
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/InviteNewMembers/InviteNewMembers.tsx:15]:
```suggestion
  const { projectID } = useParams<ProjectPathParams>();
```

in routes folder make a RouteInterfaces.ts
and add 
```
export type ProjectPathParams {
projectID: string
}
```

File: chaoscenter/web/src/controllers/InviteNewMembers/helper.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import type { User } from '@api/auth';
+import type { InviteUserDetails } from './types';
+
+export function generateInviteUsersTableContent(userData: Array<User> | undefined): Array<InviteUserDetails> {
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/InviteNewMembers/helper.ts:4]: These checks can be moved inline...this function will only confuse and this practice may not be followed by other devs if they don't know this function exists

File: chaoscenter/web/src/controllers/ProjectMember/ProjectMembers.tsx [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import ProjectMembersView from '@views/ProjectMembers';
+
+export default function ProjectMembersController(): React.ReactElement {
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/controllers/ProjectMember/ProjectMembers.tsx:4]: unnecessary controller

File: chaoscenter/web/src/views/InviteNewMembers/InviteNewMembers.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        <Icon name="cross" style={{ cursor: 'pointer' }} size={18} onClick={() => handleClose()} />
+      </Layout.Horizontal>
+      <Container width={'100%'}>{searchInput}</Container>
+      <Layout.Vertical style={{ minHeight: 150 }} flex={{ alignItems: 'center', justifyContent: 'center' }}>
+        <Loader
+          loading={isLoading}
+          noData={{
+            when: () => data.length === 0,
+            messageTitle: 'No users available',
+            message: 'No users available to send invitation'
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteNewMembers.tsx:38]: strings from yaml

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+        id: 'threeDotMenu',
+        disableSortBy: true,
+        Cell: ({ row: { original: data } }: { row: Row<InviteUserDetails> }) => {
+          const { projectID } = useParams<{ projectID: string }>();
+          const { showSuccess } = useToaster();
+          const { mutate: sendInvitationMutation, isLoading } = useSendInvitationMutation(
+            {},
+            {
+              onSuccess: () => {
+                showSuccess('Invitation sent successfully');
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:48]: strings

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+                    })
+                  }
+                />
+                <SplitButtonOption
+                  text="Viewer"
+                  onClick={() =>
+                    sendInvitationMutation({
+                      body: {
+                        projectID: projectID,
+                        role: 'Viewer',
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:89]: enum

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+                  position: PopoverPosition.BOTTOM_RIGHT
+                }}
+              >
+                <SplitButtonOption
+                  text="Editor"
+                  onClick={() =>
+                    sendInvitationMutation({
+                      body: {
+                        projectID: projectID,
+                        role: 'Editor',
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:77]: enum

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+          );
+
+          return (
+            <Layout.Vertical
+              flex={{ justifyContent: 'center', alignItems: 'flex-end' }}
+              onClick={killEvent}
+              width="100%"
+            >
+              <SplitButton
+                text="Invite as"
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:61]: strings

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+                  }
+                />
+              </SplitButton>
+            </Layout.Vertical>
+          );
+        }
+      }
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [getString]
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:102]: this dependency feels weird...we can remove this

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+      value: 'Viewer'
+    }
+  ];
+
+  const { showSuccess } = useToaster();
+  const { mutate: sendInvitationMutation, isLoading: sendLoading } = useSendInvitationMutation(
+    {},
+    {
+      onSuccess: () => {
+        showSuccess('Invitation sent successfully');
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx:92]: strings

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+      }
+    }
+  );
+
+  const { mutate: removeInvitationMutation, isLoading: removeLoading } = useRemoveInvitationMutation(
+    {},
+    {
+      onSuccess: () => {
+        showSuccess('Invitation removed successfully');
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx:101]: strings

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+    }
+  );
+
+  return (
+    <Layout.Horizontal flex={{ justifyContent: 'space-between' }} spacing="medium">
+      <Layout.Horizontal flex={{ justifyContent: 'flex-start' }} spacing="small" margin={{ bottom: 'small' }}>
+        <DropDown
+          value={memberRole}
+          items={rolesDropDown}
+          onChange={option => setMemberRole(option.label as 'Editor' | 'Owner' | 'Viewer')}
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx:112]: 'Editor' | 'Owner' | 'Viewer'
make this a type

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+          onClick={() =>
+            removeInvitationMutation({
+              body: {
+                projectID: projectID,
+                userID: data.userID
+              }
+            })
+          }
+          variation={ButtonVariation.SECONDARY}
+          text="Remove"
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx:143]: strings

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx [hunk 20]
```diff
@@ -0,0 +1,119 @@
...
+                position={Position.LEFT}
+                interactionKind={PopoverInteractionKind.HOVER}
+                disabled={data.role === 'Owner'}
+              >
+                <Button variation={ButtonVariation.ICON} icon="Options" />
+                <Menu style={{ backgroundColor: 'unset' }}>
+                  {/* <RbacMenuItem icon="edit" text="Edit Role" onClick={openEditModal} permission={PermissionGroup.OWNER} /> */}
+                  <RbacMenuItem
+                    icon="trash"
+                    text="Remove Member"
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx:74]: strings

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx [hunk 21]
```diff
@@ -0,0 +1,119 @@
...
+                  />
+                </Dialog>
+              )}
+            </Layout.Vertical>
+          );
+        }
+      }
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [getString]
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx:102]: empty deps

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx [hunk 22]
```diff
@@ -0,0 +1,119 @@
...
+  );
+  return (
+    <Layout.Vertical>
+      <Text font={{ variation: FontVariation.H6 }}>Total Members: {activeMembers.length}</Text>
+      <Loader
+        loading={isLoading}
+        noData={{
+          when: () => activeMembers.length === 0,
+          messageTitle: 'No members present',
+          message: 'No active project members available'
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ActiveMembersTable.tsx:112]: string

File: chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx [hunk 23]
```diff
@@ -0,0 +1,151 @@
...
+              value: 'Viewer'
+            }
+          ];
+
+          const { showSuccess } = useToaster();
+          const { mutate: sendInvitationMutation, isLoading: sendLoading } = useSendInvitationMutation(
+            {},
+            {
+              onSuccess: () => {
+                showSuccess('Invitation sent successfully');
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx:72]: string

File: chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx [hunk 24]
```diff
@@ -0,0 +1,151 @@
...
+                getPendingMembersRefetch();
+              }
+            }
+          );
+
+          const { mutate: removeInvitationMutation, isLoading: removeLoading } = useRemoveInvitationMutation(
+            {},
+            {
+              onSuccess: () => {
+                showSuccess('Invitation removed successfully');
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx:82]: string

File: chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx [hunk 25]
```diff
@@ -0,0 +1,151 @@
...
+                  onClick={() =>
+                    removeInvitationMutation({
+                      body: {
+                        projectID: projectID,
+                        userID: data.userID
+                      }
+                    })
+                  }
+                  variation={ButtonVariation.SECONDARY}
+                  text="Remove"
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx:125]: string

File: chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx [hunk 26]
```diff
@@ -0,0 +1,151 @@
...
+                />
+              </Layout.Horizontal>
+            </Layout.Horizontal>
+          );
+        }
+      }
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [getString]
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx:134]: empty []

File: chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx [hunk 27]
```diff
@@ -0,0 +1,151 @@
...
+  );
+  return (
+    <Layout.Vertical>
+      <Text font={{ variation: FontVariation.H6 }}>Total Pending Invitations: {pendingMembers.length}</Text>
+      <Loader
+        loading={isLoading}
+        noData={{
+          when: () => pendingMembers.length === 0,
+          messageTitle: 'No pending invitations present',
+          message: 'No pending invitations present'
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/PendingMembersTable.tsx:144]: strings

File: chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx [hunk 28]
```diff
@@ -0,0 +1,105 @@
...
+                    className={styles.toolbar}
+                    padding="medium"
+                  >
+                    <Layout.Horizontal>
+                      <RbacButton
+                        intent="primary"
+                        data-testid="add-members"
+                        icon="plus"
+                        iconProps={{ size: 10 }}
+                        text="New Member"
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx:67]: string

File: chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx [hunk 29]
```diff
@@ -0,0 +1,105 @@
...
+    <DefaultLayout breadcrumbs={[]} title={'Members'} noPadding>
+      <Container className={styles.tabContainer}>
+        <Tabs
+          id="projectMembersTabs"
+          onChange={handleTabChange}
+          selectedTabId={activeTab}
+          tabList={[
+            {
+              id: 'active-members',
+              title: 'Active Members',
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx:53]: string

File: chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx [hunk 30]
```diff
@@ -0,0 +1,105 @@
...
+                  </Layout.Horizontal>
+                  <Layout.Vertical padding="medium">
+                    <ActiveProjectMembersController />
+                  </Layout.Vertical>
+                </Layout.Vertical>
+              )
+            },
+            {
+              id: 'pending-members',
+              title: 'Pending Members',
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/ProjectMembers/ProjectMembers.tsx:93]: string

File: chaoscenter/web/src/views/RemoveMember/RemoveMember.tsx [hunk 31]
```diff
@@ -0,0 +1,55 @@
...
+export default function RemoveMemberView(props: RemoveMemberViewProps): React.ReactElement {
+  const { handleClose, username, removeMemberMutation, userID } = props;
+  const { getString } = useStrings();
+  const { projectID } = useParams<{ projectID: string }>();
+  return (
+    <Layout.Vertical padding="medium" style={{ gap: '1rem' }}>
+      <Layout.Horizontal flex={{ alignItems: 'center', justifyContent: 'space-between' }}>
+        <Text font={{ variation: FontVariation.H4 }}>Remove Member</Text>
+      </Layout.Horizontal>
+      <Text font={{ variation: FontVariation.BODY }}>Are you sure you want to remove {username}?</Text>
```
Comment (arkajyotiMukherjee) [inline chaoscenter/web/src/views/RemoveMember/RemoveMember.tsx:30]: string

**Review by arkajyotiMukherjee — COMMENTED** [review]
make sure the strings are flattened out and not nested.

File: chaoscenter/web/src/api/entities/project.ts (hunk 1 above)
Comment (hrishavjha) [reply chaoscenter/web/src/api/entities/project.ts:24]: Types are automatically generated from swagger file, these are ig old ones.

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx (hunk 2 above)
Comment (Saranya-jena) [reply chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx:14]: These are also generated by swagger, typing is not possible.

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx (hunk 12 above)
Comment (Saranya-jena) [reply chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:89]: This is already an enum generated by swagger.
image

File: chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx (hunk 13 above)
Comment (Saranya-jena) [reply chaoscenter/web/src/views/InviteNewMembers/InviteUsersTable.tsx:77]: same

File: chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx (hunk 18 above)
Comment (Saranya-jena) [reply chaoscenter/web/src/views/ProjectMembers/ActiveMembersListColumns.tsx:112]: This is a type generated by swagger: 

image

File: chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx (hunk 2 above)
Comment (Saranya-jena) [reply chaoscenter/web/src/controllers/ActiveProjectMemberList/ActiveProjectMembers.tsx:14]: `state` is already having defined values, so we can't create different types.

**Review by arkajyotiMukherjee — APPROVED** [review]

**Pull request merged**